### PR TITLE
ci: add sync-files.yaml

### LINKED ISSUE #1: Add CI
I believe we can apply almost the same settings as autoware_common.
https://github.com/autowarefoundation/autoware_common/tree/main/.github

### PR DESCRIPTION
**Pull request by kenji-miyake**
Resolves #1

### CODE REVIEW

File: .github/sync-files.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - source: .markdownlint.yaml
+    - source: .pre-commit-config-optional.yaml
+    - source: .prettierignore
+    - source: .prettierrc.yaml
+    - source: .yamllint.yaml
+
+- repository: autowarefoundation/autoware_common
+  files:
+    - source: .github/workflows/build-and-test.yaml
+    - source: .github/workflows/build-and-test-differential.yaml
```
Comment (kenji-miyake) [inline .github/sync-files.yaml:30]: @xmfcx Could you confirm the contents? :pray:

Comment (kenji-miyake) [reply .github/sync-files.yaml:30]: I removed `self-hosted` and `docs` from https://github.com/autowarefoundation/autoware.core/blob/3e192dffcefd76c3b5f31a20535b381758b16308/.github/sync-files.yaml.

Comment (xmfcx) [reply .github/sync-files.yaml:30]: I think we can start with these, we can modify as we work with the repository.

**Review by xmfcx — APPROVED** [review]

**Comment by kenji-miyake** [comment]
FYI @esteve This is how we sync files between repositories.
See also: https://github.com/autowarefoundation/autoware.core/pull/4#issuecomment-1066036556

**Pull request merged**